cdci_data_analysis/analysis/instrument.py

```diff
@@ -139,41 +142,63 @@ def set_pars_from_dic(self, par_dic, verbose=False):
 
             # this is required because in some cases a parameter is set without a name (eg UserCatalog),
             # or they don't have to set (eg scw_list)
             if par.name is not None and par.name not in params_not_to_be_included:
                 # set the value for par to a default format,
                 # or to a default value if this is not included within the request
-                updated_par_dic[par.name] = par.set_value_from_form(par_dic, verbose=verbose)
+                updated_arg_dic[par.name] = par.set_value_from_form(arg_dic, verbose=verbose)
 
                 if par.units_name is not None:
                     if par.default_units is not None:
-                        updated_par_dic[par.units_name] = par.default_units
+                        updated_arg_dic[par.units_name] = par.default_units
                     else:
                         raise InternalError("Error when setting the parameter %s: "
                                             "default unit not specified" % par.name)
                 if par.par_format_name is not None:
                     if par.par_default_format is not None:
-                        updated_par_dic[par.par_format_name] = par.par_default_format
+                        updated_arg_dic[par.par_format_name] = par.par_default_format
                     else:
                         raise InternalError("Error when setting the parameter %s: "
                                             "default format not specified" % par.name)
                 else:
                     self.logger.warning("units_name for the parameter %s not specified", par.name)
 
             self.logger.info("after normalizing, set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
-                             par, par.name, par.value, par_dic.get(par.name, None))
+                             par, par.name, par.value, arg_dic.get(par.name, None))
 
             if par.name == "scw_list":
                 self.logger.info("set_pars_from_dic>> scw_list is %s", par.value)
 
-        return updated_par_dic
+        known_argument_names = ['instrument', 
+                                'query_status', 
+                                'query_type', 
+                                'product_type', 
+                                'session_id', 
+                                'token',
+                                'api',
+                                'oda_api_version',
+                                'off_line',
+                                'job_id',
+                                'async_dispatcher',
+                                'allow_unknown_args'] + self.get_arguments_name_list()
+        self.unknown_arguments_name_list = []
+        for k in list(updated_arg_dic.keys()):
+            if k not in known_argument_names:
+                if not (self.allow_unknown_arguments or arg_dic.get('allow_unknown_args', 'False') == 'True'):
```

Comment:
Why can the `allow_unknown_args` arg be found in the input dic `arg_dic`?

Can the `allow_unknown_args` member be assigned directly from the `arg_dic, if present?

Reply:
I made two ways of allowing undefined arguments: either instrument-wide by self.allow_unknown_arguments or by passing allow_unknown_args=True in request. The former is not used, though. 

> Can the `allow_unknown_args` member be assigned directly from the `arg_dic` entry, if present?
I'm not sure, I understand your question

Reply:
> > Can the `allow_unknown_args` member be assigned directly from the `arg_dic` entry, if present?
> I'm not sure, I understand your question

Basically:
```suggestion
                self.allow_unknown_arguments = arg_dic.get('allow_unknown_args', 'False')
```
during the set_pars_from_dic method